src/audio/CMakeLists.txt

```diff
@@ -128,6 +131,7 @@ set(eq-fir_sources eq_fir/eq_fir.c eq_fir/fir.c)
 set(eq-iir_sources eq_iir/eq_iir.c eq_iir/iir.c)
 set(dcblock_sources dcblock/dcblock.c dcblock/dcblock_generic.c)
 set(crossover_sources crossover/crossover.c crossover/crossover_generic.c)
+set(tdfb_sources tdfb/tdfb.c tdfb/tdfb_generic.c eq_fir/fir.c)
```

Comment:
The FIR needs to be moved to library similarly as @johnylin76 did for IIR.

Reply:
I think we need to add conditional linking of targets.

Something like in kconfig
define beamformer
    depends on FIR

in CMAKE
if beamformer
  link FIR with beamformer

Reply:
There's a no an implementation for this idea. Review is welcome for this part!